spd/eval.py

```diff
@@ -755,6 +756,165 @@ def compute(self) -> Mapping[str, float]:
         return {"loss/faithfulness": loss.item()}
 
 
+class GeometricSimilarityComparison(StreamingEval):
+    SLOW = True
+
+    def __init__(self, model: ComponentModel, config: Config, **kwargs: Any):
+        self.model = model
+        self.config = config
+        self.reference_run_path = kwargs.get("reference_run_path")
+        if self.reference_run_path is None:
+            raise ValueError("reference_run_path is required for GeometricSimilarityComparison")
+        self.kwargs = kwargs
```

Comment:
I don't like this. I'd specifically save the things you need. E.g. density threshold